src/esp/sensor/PinholeCamera.cpp

```diff
@@ -91,6 +96,9 @@ bool PinholeCamera::getObservation(gfx::Simulator& sim, Observation& obs) {
 
   renderTarget().renderExit();
 
+  if (is_webgl_build_) {
```

Comment:
You can use `#ifdef CORRADE_TARGET_EMSCRIPTEN` here instead.

Reply:
IMO, this would be better if we have to use this check at multiple places. Also, if we find a better way to do this then final change will need to happen only at one place.

Reply:
In general, you should use `#ifdef` where you can as that signifies that whether or not this code runs is determined at compile time, which is both nice for your reader and for the compiler (as it can trivially remove that code).  If we were requiring c++17, we could use `if constexpr`, so the syntax to do `#ifdef` will eventually get cleaner :)

If you feel very strongly about `is_webgl_build_`, we should at least do

```
const bool is_webgl_build_ = 
#ifdef CORRADE_TARGET_EMSCRIPTEN
    true;
#else
    false;
#endif
```

So that the compiler can, hopefully, optimize out the if statements (we should avoid adding unnecessary jumps to something so core as getting observations).

With all this said, it does seem weird that the sensor is responsible for calling `renderTarget().blitRgbaToDefault()`. Maybe add either an argument to `SimulatorWithAgents::getAgentObservation` to do the blit or add a `SimulatorWithAgents::blitRgbaToDefault(int agentId, const std::string& sensorId)` method and call that from JS?

Reply:
SimulatorWithAgents::blitRgbaToDefault sounds reasonable. Maybe a different name though like displayObservation?. It would also be nice to avoid the readFrameRgba in the blit path since its expensive and the data isn't used.

Reply:
Thanks both. The overall motivation of Erik's original comment is much more clear now. I will soon push a fix.